cmd/root.go

```diff
@@ -26,64 +26,75 @@ type Scanner interface {
+// list of supported format of logs to stdout & stderr
+var logFormatSupportedFormats = [1]string{"json"}
 
 // list of namespaces to be skipped from scan
 var skippedNs []string
 
 // skip SSL cert validation when connecting to PolicyServers endpoints
 var insecureSSL bool
 
-var (
-	// rootCmd represents the base command when called without any subcommands
-	rootCmd = &cobra.Command{
-		Use:   "audit-scanner",
-		Short: "Reports evaluation of existing Kubernetes resources with your already deployed Kubewarden policies",
-		Long: `Scans resources in your kubernetes cluster with your already deployed Kubewarden policies.
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "audit-scanner",
+	Short: "Reports evaluation of existing Kubernetes resources with your already deployed Kubewarden policies",
+	Long: `Scans resources in your kubernetes cluster with your already deployed Kubewarden policies.
 Each namespace will have a PolicyReport with the outcome of the scan for resources within this namespace.
 There will be a ClusterPolicyReport with results for cluster-wide resources.`,
 
-		RunE: func(cmd *cobra.Command, args []string) error {
-			level.SetZeroLogLevel()
-			namespace, err := cmd.Flags().GetString("namespace")
-			if err != nil {
-				return err
-			}
-			kubewardenNamespace, err := cmd.Flags().GetString("kubewarden-namespace")
-			if err != nil {
-				return err
-			}
-			clusterWide, err := cmd.Flags().GetBool("cluster")
-			if err != nil {
-				return err
-			}
-			policyServerURL, err := cmd.Flags().GetString("policy-server-url")
-			if err != nil {
-				return err
-			}
-			caCertFile, err := cmd.Flags().GetString("extra-ca")
-			if err != nil {
-				return err
-			}
-
-			policiesFetcher, err := policies.NewFetcher(kubewardenNamespace, skippedNs)
-			if err != nil {
-				return err
-			}
-			resourcesFetcher, err := resources.NewFetcher(kubewardenNamespace, policyServerURL)
-			if err != nil {
-				return err
-			}
-			scanner, err := scanner.NewScanner(policiesFetcher, resourcesFetcher, printJSON, insecureSSL, caCertFile)
-			if err != nil {
-				return err
-			}
-
-			return startScanner(namespace, clusterWide, scanner)
-		},
-	}
-)
+	RunE: func(cmd *cobra.Command, args []string) error {
+		level.SetZeroLogLevel()
+		namespace, err := cmd.Flags().GetString("namespace")
+		if err != nil {
+			return err
+		}
+		kubewardenNamespace, err := cmd.Flags().GetString("kubewarden-namespace")
+		if err != nil {
+			return err
+		}
+		clusterWide, err := cmd.Flags().GetBool("cluster")
+		if err != nil {
+			return err
+		}
+		policyServerURL, err := cmd.Flags().GetString("policy-server-url")
+		if err != nil {
+			return err
+		}
+		caCertFile, err := cmd.Flags().GetString("extra-ca")
+		if err != nil {
+			return err
+		}
+		logFormat, err := cmd.Flags().GetString("log-fmt")
+		if err != nil {
+			return err
+		}
+		switch logFormat {
+		// TODO use slices package in go 1.21
```

Comment:
I think we can start to use it already. In the CA changes PR I've updated the go to 1.21 as well.

Reply:
Sure, let's move to Go 1.21

Reply:
done!